### PR TITLE
prime result ids on search rest endpoint

### PR DESCRIPTION
**Pull request by manuelRod**
Trac ticket: 55674

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/rest-api/rest-search-controller.php [hunk 1]
```diff
@@ -332,6 +332,25 @@ public function test_get_items_search_for_foocontent() {
...
+		// 'foocontent' was searched in test_get_items_search_for_foocontent, ids should be primed already.
+		$response  = $this->do_request_with_params(
+			array(
+				'per_page' => 100,
+				'search'   => 'foocontent',
+			)
+		);
+		$args      = $action->get_args();
+		$last_args = end( $args );
+		$this->assertSameSets( wp_list_pluck( $response->get_data(), 'id' ), $last_args[1], 'Ensure that post ids are primed' );
```
Comment (spacedmonkey) [inline tests/phpunit/tests/rest-api/rest-search-controller.php]:
```suggestion
                 $ids = wp_list_pluck( $response->get_data(), 'id' );
		$this->assertSameSets( $ids, $last_args[1] );
```

File: src/wp-includes/rest-api/search/class-wp-rest-post-search-handler.php [hunk 2]
```diff
@@ -84,6 +84,9 @@ public function search_items( WP_REST_Request $request ) {
 
 		$query     = new WP_Query();
 		$found_ids = $query->query( $query_args );
+		if ( ! empty( $found_ids ) ) {
+			_prime_post_caches( $found_ids, true, true );
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api/search/class-wp-rest-post-search-handler.php]:
```suggestion
			_prime_post_caches( $found_ids, false, false );
```

Do we need to prime meta and terms here?

Comment (spacedmonkey) [reply src/wp-includes/rest-api/search/class-wp-rest-post-search-handler.php]: Or maybe 
```suggestion
			_prime_post_caches( $found_ids, false, true );
```

**Comment by manuelRod** [comment]
@spacedmonkey take a look at the new test, if ids are primed, that indeed would be the last query.

File: tests/phpunit/tests/rest-api/rest-search-controller.php [hunk 3]
```diff
@@ -332,6 +352,28 @@ public function test_get_items_search_for_foocontent() {
 		);
 	}
 
+	/**
+	 * @ticket 55674
+	 * Make sure post ids are primed.
```
Comment (spacedmonkey) [inline tests/phpunit/tests/rest-api/rest-search-controller.php]: Comments like this are not needed.

File: tests/phpunit/tests/rest-api/rest-search-controller.php [hunk 4]
```diff
@@ -332,6 +352,28 @@ public function test_get_items_search_for_foocontent() {
...
+	 */
+	public function test_get_items_search_prime_ids() {
+		global $wpdb;
+
+		$query_args = array(
+			'per_page' => 100,
+			'search'   => 'primedpoststitles',
+		);
+		$response   = $this->do_request_with_params( $query_args );
+		$this->assertSame( 200, $response->get_status() );
```
Comment (spacedmonkey) [inline tests/phpunit/tests/rest-api/rest-search-controller.php]: Tests with multiple asserts should really have a message added. See https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions

File: src/wp-includes/rest-api/search/class-wp-rest-post-search-handler.php [hunk 5]
```diff
@@ -84,6 +84,9 @@ public function search_items( WP_REST_Request $request ) {
 
 		$query     = new WP_Query();
 		$found_ids = $query->query( $query_args );
+		if ( ! empty( $found_ids ) ) {
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api/search/class-wp-rest-post-search-handler.php]: Another solution might be to do this.

```
$query_args = array(
			'post_type'           => $post_types,
			'post_status'         => 'publish',
			'paged'               => (int) $request['page'],
			'posts_per_page'      => (int) $request['per_page'],
			'ignore_sticky_posts' => true,
			'update_post_term_cache' => false, 
                        'update_post_meta_cache' => false,
		);

$query     = new WP_Query();
$posts = $query->query( $query_args );
$found_ids = wp_list_pluck( $posts, 'ID' );
```

The benefit of this, it more likely to go through caching plugins.

**Comment by manuelRod** [comment]
@spacedmonkey ready, let me know if all good and i open the PR in core.

**Review by spacedmonkey — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**